openapi-model/src/main/scala/sttp/apispec/openapi/validation/OpenAPIComparator.scala

```diff
@@ -0,0 +1,284 @@
+package sttp.apispec.openapi.validation
+
+import sttp.apispec.{Schema, SchemaLike}
+import sttp.apispec.openapi.{Header, MediaType, OpenAPI, Operation, Parameter, PathItem, RequestBody, Response}
+import sttp.apispec.validation.SchemaComparator
+
+import scala.collection.immutable.ListMap
+
+class OpenAPIComparator(
+    clientOpenAPI: OpenAPI,
+    serverOpenAPI: OpenAPI
+) {
+  private val httpMethods = List("get", "post", "patch", "delete", "options", "trace", "head", "put")
+  private var serverSchemas: Map[String, Schema] = Map.empty[String, Schema]
```

Comment:
we definitely shouldn't have mutable state that's shared among a couple of method invocations. This makes an `OpenAPIComparator` instance not re-usable. We either need to pass the state explicitly, or create a private mutable comparator instance which will be single-use

Reply:
Let's add a test that you can reuse the comparator instance